notes/design_principles.md

```diff
@@ -0,0 +1,101 @@
+
+## Design Principles for Data Modeling in HyBox
+
+### Use discovery and search requirements as a pragmatic means to determine the needed extent of the modeling efforts
```

Comment:
>  If there are real use cases for additional metadata specificity, lets get them in as feedback and respond then, rather than over-engineering from day zero.

:+1: I'd tag @guegueng @ggeisler and @hannahfrost here, but I think this sounds fine for now.